### ISSUE TITLE
Move Focus to Quick_Find search-box Upon Typing

### ISSUE THREAD

**Issue by Lonniebiz** [FreeRDP/Remmina#837]
I log into and out of numerous machines each day, and I'm constantly having to click focus into the Quick_Find text box after giving focus to the window itself.

Since typing does nothing, when the Quick_Find search-box is NOT in focus, it would be very nice if it could gain focus in the event that I start typing. Even if it loses my first keystroke, it would be very nice if hitting any key would make it gain focus. This is my most fundamental request. If you want to get fancy you can implement this feature in way that is does not lose any keystokes (meaning you can start typing full speed and the Quick_Find search-box not only gains focus, but it inserts all of your keystrokes -- no matter how fast you type them).

However, bare minimum, it would be very nice if any keystoke would automatically give focus to the Quick_Find search-box.

**Comment by antenore** [comment]
I start to work on this in the remmina_main #875 , but weirdly it dosn't work.

In the meanwhile, would you like to cast your vote regarding the new interface?

https://plus.google.com/113261751308618587686/posts/98PazVJ82ZG

**Comment by GR0SZ-ek** [comment]
It can combine the functionality window "quick search" and "fast connect"? When entering an address would be searchable database connections.

**Comment by giox069** [comment]
@GR0SZ-ek: yes, the new textbox on the new remmina main toolbar already works as both "quick search" and "fast connect".

**Comment by giox069** [comment]
You can follow the discussion on the new remmina_main here:
https://github.com/FreeRDP/Remmina/pull/875

**Comment by antenore** [comment]
closed by  #875

**Comment by Lonniebiz** [comment]
Any chance we can open this one back up.

My goal is to switch to the Remmina window and start typing (without having to click into the search box first). **Not just on first launch, but on every successive time I switch to that window**.

I propose that the best way to do this, is to put a few lines of code into the method/function that processes the double-click event that launches a remote session. When the even occurs, return focus and highlighting to the search box. This way, every time you return to the window to look up another remote desktop, the window is ready for you to start typing; it will overwrite your previous search upon the first keystroke.

**Comment by antenore** [comment]
I'll try to work this out in the next days

**Comment by Lonniebiz** [comment]
Thanks man. I tried looking through the source code, but I couldn't figure out where this would go.

**Comment by antenore** [comment]
99% is this file https://github.com/FreeRDP/Remmina/blob/next/remmina/ui/remmina_main.glade you can edit by hand or with glade

To do some tests you can even edit the installed one, normally under /opt/remmina_devel/remmina/share/remmina/ui/remmina_main.glade (or /usr/local/share... or /usr/share/...)

I've already tried something but it didn't work as expected, I'm missing something or I didn't understand the doc.

**Comment by antenore** [comment]
A quick note, pressing ctrl+f  will bring the cursor in the search box.

**Comment by Lonniebiz** [comment]
Line 76 indicates the event-handler we're looking for (I think), but where is the actual code that gets fired? The same code that fires when you press ctrl-F, should also fire during the remmina_main_on_action_connection_connect event it seems. Then, the only remaining line(s) of code would be to highlight the text after the focus occurs.

**Comment by antenore** [comment]
:-\ didn't have too much time to look at it, these months I've a very limited free time.

So far so bad, all the tests I've done were failing.

**Comment by Lonniebiz** [comment]
Well, I do appreciate you trying. If I'd familiarize myself with Remmina's coding environment/languages, I could help out. My strength is in javascript and c#, so I've got to figure out how to get started with whatever Remmina is.
